replica/database.cc

```diff
@@ -1012,19 +1018,11 @@ future<> database::add_column_family(keyspace& ks, schema_ptr schema, column_fam
     }
     // avoid self-reporting
     auto& sst_manager = is_system_table(*schema) ? get_system_sstables_manager() : get_user_sstables_manager();
-    lw_shared_ptr<column_family> cf;
-    if (cfg.enable_commitlog && _commitlog) {
-        db::commitlog& cl = schema->static_props().use_schema_commitlog && _uses_schema_commitlog
-                ? *_schema_commitlog
-                : *_commitlog;
-        cf = make_lw_shared<column_family>(schema, std::move(cfg), ks.metadata()->get_storage_options_ptr(), cl, _compaction_manager, sst_manager, *_cl_stats, _row_cache_tracker, erm);
-    } else {
-       cf = make_lw_shared<column_family>(schema, std::move(cfg), ks.metadata()->get_storage_options_ptr(), column_family::no_commitlog(), _compaction_manager, sst_manager, *_cl_stats, _row_cache_tracker, erm);
-    }
+    auto cf = make_lw_shared<column_family>(schema, std::move(cfg), ks.metadata()->get_storage_options_ptr(), _compaction_manager, sst_manager, *_cl_stats, _row_cache_tracker, erm);
     cf->set_durable_writes(ks.metadata()->durable_writes());
 
     if (!readonly) {
-        cf->mark_ready_for_writes();
+        cf->mark_ready_for_writes(commitlog_for(schema));
```

Comment:
The precondition for calling `commitlog_for` is that `_uses_schema_commitlog` has been set to the correct value -- i.e., we shouldn't call it before `maybe_init_schema_commitlog()`, is that right? So it's incorrect to call `add_column_family(..., false)` before `maybe_init_schema_commitlog()`?

Maybe we should protect ourselves better by making `_uses_schema_commitlog` an `std::optional`, initialize it to `nullopt` and only set it when we make the decision whether we make the decision to use schema commitlog or not after enabling features. In `commitlog_for` we could then verify that it's set and catch bugs.

Reply:
good point, will update the code, thanks!